resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -402,5 +402,10 @@ public void abortIfClosed()
    {
        if (bufferedEntity == null) super.abortIfClosed();
    }
+   
+   public void closeHttpResponse()
+   {
+        

```

Comment:
This should probably be an abstract method (also considering the ClientResponse class here is abstract). A bit of javadoc would have to be added, so that implementors know what do in their classes extending this.

Reply:
Personnaly I'm not really comfortable with having both `close()` , `closeHttpResponse()` and `releaseConnection()` methods. 
IMO from the implementators point of view `closeHttpResponse()` and  `releaseConnection()` are pretty similar. So why not just having a `releaseConnection(boolean consumeInputStream)` method ? The concept of `boolean consumeInputStream` is common to both Apache HTTP Client and JDK HttpUrlConnection so it will not be that disturbing for implementators.
We will have to keep existing `releaseConnection()` for compatibility reason but it will just invoke `releaseConnection(true)`. 
WDYT ?

Reply:
Hi @asoldano , @jimma 

To move forward, I created a PR with the proposal I was talking about above: https://github.com/resteasy/Resteasy/pull/1534

Reply:
@NicoNes sorry for the late reply, do you mean you're proposing this PR to be closed and replaced with #1534 ?

Reply:
Hi @asoldano,

Right, you can close this one. I've merged @jimma changes from this PR to the new one plus the changes I was talking about in the comment above.